ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/range/RangeHeaders.java

```diff
@@ -50,7 +41,7 @@ public List<BlockHeader> getHeadersToImport() {
   }
 
   public BlockHeader getFirstHeaderToImport() {
-    return headersToImport.get(0);
+    return headersToImport.size() > 0 ? headersToImport.get(0) : null;
```

Comment:
Making this return Optional might be a bit more helpful to the reader, and would lend itself to simpler conversion to an empty stream later on.